runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -11960,7 +11960,7 @@ void VMgenerateCatchBlockBBStartPrologue(TR::Node *node, TR::Instruction *fenceI
       TR::Register *biAddrReg = cg->allocateRegister();
       TR::Register *recompCounterReg = cg->allocateRegister();
       intptr_t addr = (intptr_t) (comp->getRecompilationInfo()->getCounterAddress());
-      TR::Instruction *cursor = loadAddressConstant(cg, comp->compileRelocatableCode(), node, addr, biAddrReg);
+      TR::Instruction *cursor = loadAddressConstant(cg, comp->compileRelocatableCode(), node, addr, biAddrReg, NULL, false, TR_BodyInfoAddressLoad);
```

Comment:
This query needs to be `cg->needRelocationsForBodyInfoData()` like the other places where we generate relocation records for the recompilation counter (mostly in OMR). The `comp->compileRelocatableCode()` query is wrong because we still need to create records when compiling non-AOT methods at the JITServer, of course.